### PR TITLE
Remove fmt and vet checks from CI

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

- golangci-lint already does the vet and fmt checks. Remove those
as it is redundant now.
- Also start running golangci-lint as part of the CI pipeline now
just as an extra layer of reliability.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -154,6 +154,8 @@ govet: ## Runs govet against all packages.
 	env GO111MODULE=off $(GO) get golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow
 	$(GO) vet $(GOFLAGS) $(ALL_PACKAGES) || exit 1
 	$(GO) vet -vettool=$(GOPATH)/bin/shadow $(GOFLAGS) $(ALL_PACKAGES) || exit 1
+
+checker:
```
Comment (hanzei) [inline Makefile]: Can you please rename it to 
```suggestion
plugin-checker:
```

File: Makefile [hunk 2]
```diff
@@ -173,11 +175,11 @@ gofmt: ## Runs gofmt against all packages.
...
 	@echo "gofmt success"; \
 
-golangci-lint: ## Run golangci-lint on codebasis
+golangci-lint: ## Run golangci-lint on codebase
 # https://stackoverflow.com/a/677212/1027058 (check if a command exists or not)
 	@if ! [ -x "$$(command -v golangci-lint)" ]; then \
-		echo "golangci-lint is not installed. Please see https://github.com/golangci/golangci-lint#install for installation instructions."; \
-		exit 1; \
+		echo "Installing golangci-lint"; \
+		curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(GOPATH)/bin v1.21.0; \
```
Comment (hanzei) [inline Makefile]: I still don't think we should install software without the permission of the user. Would you please add the installation to the circle ci build?

File: Makefile [hunk 3]
```diff
@@ -229,8 +231,7 @@ check-licenses: ## Checks license status.
 check-prereqs: ## Checks prerequisite software status.
 	./scripts/prereq-check.sh
 
-# TODO: remove govet and gofmt checks once golangci-lint is being enforced.
-check-style: govet gofmt check-licenses check-plugin-golint ## Runs govet and gofmt against all packages and also ensures plugin package golint compliant
+check-style: golangci-lint checker check-licenses check-plugin-golint ## Runs golangci against all packages and also ensures plugin package golint compliant
```
Comment (hanzei) [inline Makefile]: We can remove `gofmt` and `govet` now, right?

Comment (agnivade) [reply Makefile]: Yeah, it _is_ removed. Did I misunderstand something ?

File: Makefile (hunk 2 above)
Comment (agnivade) [reply Makefile]: Oh you want me to do it in the circleCI setup, sure.

Comment (agnivade) [reply Makefile]: So ideally, I should do all the tool installations in the `setup` job in circleCI. But the installed tool needs to persist to be used in the build job. Now only `/go/src/github.com/mattermost` and its descendants are persisted, which means I have to place the binary somewhere there. Or alternatively download the tool in the `build` job itself.

What do you suggest ?

On a related note: I suggest that the tool be installed in the docker image itself. That should make the build much faster. And we just need to update the image once in a while when a new release comes out.

File: Makefile (hunk 3 above)
Comment (hanzei) [reply Makefile]: I mean removing the Makefile targets completely.

File: Makefile (hunk 2 above)
Comment (hanzei) [reply Makefile]: Updating the docker image requires currently some more work. Having it in the CI should be fine for now.

I'm totally fine with installing it in `build`. We only use it there.

**Comment by agnivade** [comment]
I don't understand the jenkins failures. The first time it did not execute the curl command at all. The next time, it is saying ` cd: can't cd to /go/src/github.com/mattermost/mattermost-server`

**Comment by agnivade** [comment]
Aha :bulb:

**Comment by agnivade** [comment]
Sorry but I must be missing something. It is simply not executing the curl command at all.

```
+ cd /go/src/github.com/mattermost/mattermost-server
+ make config-reset
Resetting configuration to default
rm -f config/config.json
OUTPUT_CONFIG=/go/src/github.com/mattermost/mattermost-server/config/config.json go generate  -mod=vendor ./config
+ make check-style BUILD_NUMBER=PR-13063-7
golangci-lint is not installed. Please see https://github.com/golangci/golangci-lint#install for installation instructions.
make: *** [Makefile:157: golangci-lint] Error 1
script returned exit code 2
```

There are these 2 lines in between which are simply being skipped somehow

```
echo "Installing golangci-lint"
curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b /usr/local/bin v1.21.0
```

**Comment by hanzei** [comment]
@cpanato Can you please help with the jenkins failure? I don't know why this is happening.

**Comment by cpanato** [comment]
this PR adds the golangci thing only for circleci need to add the same for jenkins @hanzei

**Comment by hanzei** [comment]
It does it to jenkins: https://github.com/mattermost/mattermost-server/pull/13063/files#diff-4fcbdb7c36cf4423a38ecbe32272e622

Or am I misunderstanding?

**Comment by cpanato** [comment]
@hanzei since this comes from a contributor and not from an org member Jenkins will not update the pipeline, only when we merge this PR

**Comment by hanzei** [comment]
Well, then let's hope this will work :wink:

File: Makefile [hunk 4]
```diff
@@ -229,8 +208,7 @@ check-licenses: ## Checks license status.
 check-prereqs: ## Checks prerequisite software status.
 	./scripts/prereq-check.sh
 
-# TODO: remove govet and gofmt checks once golangci-lint is being enforced.
-check-style: govet gofmt check-licenses check-plugin-golint ## Runs govet and gofmt against all packages and also ensures plugin package golint compliant
+check-style: golangci-lint plugin-checker check-licenses check-plugin-golint ## Runs golangci against all packages and also ensures plugin package golint compliant
```
Comment (hanzei) [inline Makefile]: We can also remove the `megacheck` command. It is broken anyway.

Comment (agnivade) [reply Makefile]: Yep, saw that. I will do it in the PR which deals with the tools dependency since megacheck is a tool.

**Comment by agnivade** [comment]
> Should we mention/document golangci-lint installation in the developer setup?

I think that's a good idea.

> Does golangci-lint works fine on Windows (for those who run local dev in that platform)?

It is completely written in Go, and hence I believe should work on Windows.

**Comment by hanzei** [comment]
@agnivade Heads up that `golangci-lint` won't check the enterprise code. Do you want to submit a PR for this or should I?

**Comment by hanzei** [comment]
@cpanato How do we merge this? The jenkis check is required, but I suppose the build will be green on `master`.

**Comment by cpanato** [comment]
we merge, if the Jenkins broke for other PR we revert or fix @hanzei

**Comment by hanzei** [comment]
@cpanato I can't merge. Can you?

Screenshot from 2019-11-18 12-07-52

**Comment by hanzei** [comment]
Will check if jenkis runs fine.

**Comment by agnivade** [comment]
>  Do you want to submit a PR for this or should I?

I can do it.

**Pull request merged**